### ISSUE TITLE
Add IsNullOrEmpty or IsNullOrWhiteSpace check for method parameters throws an empty "message"

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#41822]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/922068/add-isnullorempty-or-isnullorwhitespace-check-for.html)._

---
If you use ctrl + . on a string method parameter to generate IsNullOrEmpty or IsNullOrWhiteSpace checks the resulting exception generated is not ideal and looks like (when the parameter name is "path" as an example):

```
if (string. IsNullOrEmpty(path))
    throw new ArgumentException("message", nameof(path));
```

or 

```
if (string. IsNullOrWhiteSpace(path))
    throw new ArgumentException("message", nameof(path));
```

The problem is the "message" that is added is just "message" but it should be something descriptive like 

`$"{nameof(path)} cannot be null or empty"`

---
#### Visual Studio Feedback System on 2/18/2020, 11:28 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by sharwell** [comment]
It seems reasonable to generate this:

```
$"'{nameof(path)}' cannot be null or empty"
```

**Comment by gio-frigione** [comment]
Hi @sharwell, I want to start contributing to the roslyn project and I think this issue could be a good candidate to start getting used to the whole process. 

I just have couple of questions that I'd like to clear before submitting a pull request:

- Do you think we need to differentiate the message for the ArgumentException generated by IsNullOrEmpty() from the one generated by IsNullOrWhitespace() or the "empty" semantic is clear from the context?
- Am I correct in assuming that the message needs to be localized? If yes, I think I need to add an entry to the FeaturesResources.resx. Just out of curiosity, is there a place where I can read about how the whole translation process of these resources work?

Thank you.

**Comment by Sliptory** [comment]
The problem is this:
In the form in which I did it now (you can look at the commit), this does not work in VB (tests do not pass). What to do with this, I do not understand yet.

**Comment by CyrusNajmabadi** [comment]
>  this does not work in VB (tests do not pass

Can you open a PR so someone can pull this locally to try it out?

**Comment by Sliptory** [comment]
> > this does not work in VB (tests do not pass
> 
> Can you open a PR so someone can pull this locally to try it out?

I reopened PR

**Comment by JoeRobich** [comment]
Fixed in 16.7 Preview 1 by 1https://github.com/dotnet/roslyn/pull/42101
